src/pages/index.js

```diff
@@ -14,70 +14,174 @@ import {
   inputJob,
 } from "../utils/constants.js";
 import "./index.css";
+import Api from "../components/Api.js";
+import PopupWithVerify from "../components/PopupWithVerify.js";
 
+//Api
+const api = new Api("https://around-api.en.tripleten-services.com/v1", {
+  authorization: "57aa7463-f839-44ea-835b-493cff8bea63",
+  "Content-Type": "application/json",
+});
+
+api
+  .getInitialCards()
+  .then((res) => {
+    res.forEach((card) => {
+      newSection.appendItem(createCard(card));
+    });
+  })
+  .catch(console.error);
+
+api
+  .getUserInfo()
+  .then((res) => {
+    userInfo.setUserInfo({
+      name: res.name,
+      job: res.about,
+    });
+    userInfo.setUserAvatar(res.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+//functions
+const handleAddSubmit = (data) => {
+  addFormPopup.handleLoadingText(true);
+  api
+    .addNewCard(data.name, data.link)
+    .then((cardData) => {
+      newSection.prependItem(createCard(cardData));
+      addFormPopup.resetForm();
+      addFormPopup.close();
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      addFormPopup.handleLoadingText(false);
+    });
+};
+
+const handleEditSubmit = (inputValue) => {
+  editFormPopup.handleLoadingText(true);
+  api
+    .editProfile(inputValue.title, inputValue.description)
+    .then((data) => {
+      const profileInfo = {};
+      profileInfo.name = data.name;
+      profileInfo.job = data.about;
+      userInfo.setUserInfo(profileInfo);
+    })
+    .catch((err) => console.error(err))
+    .finally(() => {
+      editFormPopup.handleLoadingText(false);
+    });
+};
+
+const handleDeleteButton = (card) => {
+  deletePopup.open();
+  deletePopup.setHandleDeleteMethod(() => {
+    api
+      .deleteCard(card._id)
```

Comment:
Needs correction: don't forget to notify the user about loading by changing the button text

Reply:
Hi, I double-checked the project description and didn't think we need to change the button text to delete the popup?

Reply:
You should always notify the user about the loading somehow, so please change the button text to `Loading..` before the request and change it back after, like you do that for all the other requests